presto-hive/src/main/java/com/facebook/presto/hive/orc/OrcBatchPageSourceFactory.java

```diff
@@ -236,9 +235,9 @@ public static OrcBatchPageSource createOrcPageSource(
             throw new PrestoException(HIVE_CANNOT_OPEN_SPLIT, splitError(e, path, start, length), e);
         }
 
-        AggregatedMemoryContext systemMemoryUsage = newSimpleAggregatedMemoryContext();
+        HiveOrcAggregatedMemoryContext systemMemoryUsage = new HiveOrcAggregatedMemoryContext();
```

Comment:
nit: `OrcAggregatedMemoryContext`. Use interfaces whenever possible on declaration. Same for all other places.